src/Tests/StatsdSenderTests.cs

```diff
@@ -0,0 +1,630 @@
+                IList<object[]> argsPerCall = _udp.GetArgumentsForCallsMadeOn(x => x.Send(Arg<string>.Is.Anything));
+                Assert.That(argsPerCall.Count, Is.EqualTo(1));
+                Assert.That(((string)argsPerCall[0][0]), Is.EqualTo("testMetric:" + metricsToSend.ToString() + "|c"));
+            }
+
+            [Test]
+            public void aggregates_gauges()
+            {
+                var sender = new ThreadSafeConsumerProducerSender(new ThreadSafeConsumerProducerSender.Configuration() { StatsdUDP = _udp, MaxSendDelayMS = 1000 });
+                var metricsToSend = 10;
+                Metric lastMetricSent = null;
+                for (var i = 0; i < metricsToSend; i++)
+                {
+                    var metric = new Gauge() { Name = "testMetric", ValueAsDouble = 1 };
+                    sender.Send(metric);
+                    lastMetricSent = metric;
+                }
+
+                Thread.Sleep(1500);
+                IList<object[]> argsPerCall = _udp.GetArgumentsForCallsMadeOn(x => x.Send(Arg<string>.Is.Anything));
+                Assert.That(argsPerCall.Count, Is.EqualTo(1));
+                Assert.That(((string)argsPerCall[0][0]), Is.EqualTo(lastMetricSent.Command));
+            }
+
+            [Test]
+            public void does_not_aggregate_timers()
+            {
+                var sender = new ThreadSafeConsumerProducerSender(new ThreadSafeConsumerProducerSender.Configuration() { StatsdUDP = _udp, MaxSendDelayMS = 1000 });
+                var metricsToSend = 10;
+                var metric = new Timing() { Name = "testMetric", ValueAsInt = 50 };
+                for (var i = 0; i < metricsToSend; i++)
+                    sender.Send(metric);
+
+                Thread.Sleep(1500);
+                IList<object[]> argsPerCall = _udp.GetArgumentsForCallsMadeOn(x => x.Send(Arg<string>.Is.Anything));
+
+                for (var i = 0; i < argsPerCall.Count; i ++)
+                {
+                    var packet = (string)argsPerCall[i][0];
+                    var lines = packet.Split('\n');
+                    for(var j = 0; j < lines.Length; j ++)
+                    {
+                        Assert.That(lines[j], Is.EqualTo(metric.Command));
+                    }
+                }
+            }
+
+            [Test]
+            public void stops_worker_threads_after_dispose()
+            {
+                var sender = new ThreadSafeConsumerProducerSender(new ThreadSafeConsumerProducerSender.Configuration() { StatsdUDP = _udp, MaxSendDelayMS = 1000 });
+                var metric = new Timing() { Name = "testMetric", ValueAsInt = 50 };
+
+                sender.Dispose();
+                sender.Send(metric);
+
+                Thread.Sleep(1500);
+                IList<object[]> argsPerCall = _udp.GetArgumentsForCallsMadeOn(x => x.Send(Arg<string>.Is.Anything));
+                Assert.That(argsPerCall.Count, Is.EqualTo(0));
+            }
+        }
+
+/*
```

Comment:
Good catch.  I commented this out while making some changes and forgot to re-enable.  Hold on this while I update tests and commit an update to the branch..